File: Tutorial1DaggerBasics/src/main/java/com/example/tutorial1daggerbasics/MainActivity.java
package com.example.tutorial1daggerbasics;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.widget.TextView;

import com.example.tutorial1daggerbasics.model.MyExample;

import java.util.Date;

import javax.inject.Inject;

public class MainActivity extends AppCompatActivity {

    @Inject
    MyExample mMyExample;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        TextView dateTextView = findViewById(R.id.tv_date);

        ((MyApplication) getApplication())
                .getMyComponent()
                .inject(MainActivity.this);

        dateTextView.setText((new Date(mMyExample.getDate())).toString());
    }
}

File: Tutorial5-2Dagger-Android-SubComponents2SeedInstance/src/main/java/com/example/tutorial5_2dagger_android_subcomponents2/MainActivity.kt
package com.example.tutorial5_2dagger_android_subcomponents2

import android.content.Intent
import android.content.SharedPreferences
import android.os.Bundle
import android.widget.Button
import android.widget.TextView
import com.example.tutorial5_2dagger_android_subcomponents2.di.ToastMakerSubComponent
import com.example.tutorial5_2dagger_android_subcomponents2.model.ToastMaker
import dagger.android.support.DaggerAppCompatActivity
import javax.inject.Inject

/**
 * This tutorial shows how to use @Binds with seedInstance method in [ToastMakerSubComponent]
 * class to get context from a MainActivity in this example
 */
class MainActivity : DaggerAppCompatActivity() {
    // Injected from ApplicationModule
    @Inject
    lateinit var sharedPreferences: SharedPreferences
    // Injected from ToastMaker SubComponent
    @Inject
    lateinit var toastMaker: ToastMaker


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        toastMaker.showToast("sharedPreferences $sharedPreferences")

        findViewById<TextView>(R.id.tvInfo).text = "sharedPreferences: sharedPreferences\n" +
                "toastMaker: $toastMaker"


        findViewById<Button>(R.id.button).setOnClickListener {

            val intent = Intent(this@MainActivity, SecondActivity::class.java)
            startActivity(intent)

        }
    }
}